gridstatus/caiso.py

```diff
@@ -325,6 +325,10 @@ def get_lmp(
             verbose=verbose,
         )
 
+        if df.empty:
+            log(f"No data found for date: {date} and end: {end}", verbose=verbose)
```

Comment:
let's raise custom `NoDataFound`. Also use it: https://github.com/WillKoehrsen/gridstatus/blob/548e3561978407d6009c32007e927dd2da36f4c1/gridstatus/pjm.py#L898